tests/lsp_features/folding_range.zig

```diff
@@ -0,0 +1,66 @@
+const std = @import("std");
+const zls = @import("zls");
+const builtin = @import("builtin");
+
+const Context = @import("../context.zig").Context;
+
+const types = zls.types;
+const requests = zls.requests;
+
+const allocator: std.mem.Allocator = std.testing.allocator;
+
+test "foldingRange - empty" {
+    try testFoldingRange("", "[]");
+}
+
+test "foldingRange - smoke" {
+    try testFoldingRange(
+        \\fn main() u32 {
+        \\    return 1 + 1;
+        \\}
+    ,
+        \\[{"startLine":0,"endLine":1}]
```

Comment:
In rust-analyzer, we've went for something smarter and used placeholders to delimit ranges:

https://github.com/rust-lang/rust-analyzer/blob/d7be2fa1a6d8375465c026efd079f7c7f0678775/crates/ide/src/folding_ranges.rs#L368-L369

retrospectively, this wasnt' too useful for this use-case, just dumping JSON should be fine.